### PR TITLE
Update phpstan to 0.12

### PR DESCRIPTION
**Pull request by adaamz**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,23 +10,9 @@ before_install:
...
-      php: 7.2
-      install:
-        - travis_retry composer update --no-progress --prefer-dist --prefer-lowest
-      script:
-        - composer run-script tests
-
-    - stage: Quality Assurance
-      php: 7.2
-      script:
-        - composer run-script qa
```
Comment (mabar) [inline .travis.yml:29]: Why are you changing build?

Comment (adaamz) [reply .travis.yml:29]: bcs master is failing and phpstan does not run in pipeline

this pr is still wip

**Comment by f3l1x** [comment]
Could you please @adaamz repair CI?

**Comment by adaamz** [comment]
I fixed the phpstan issues, but there is some inconsistence between PHP 7.2 and PHP 7.3 BcMath. Can anybody look on that? Thanks.

**Comment by MartkCz** [comment]
Failing test fixed

@adaamz Could you fix cs, please?

https://travis-ci.org/github/contributte/invoice/jobs/610833195

**Comment by adaamz** [comment]
@MartkCz 
- rebased
- fixed issues
- updated to phpstan 0.12
- installed phpstan/extension-installer (to really use required extensions in phpstan)
- removed phpstan-strict-rules (almost all rules where voleted)

**Comment by f3l1x** [comment]
LGTM

**Comment by f3l1x** [comment]
Merged. Thx.

**Pull request merged**